book/content/boundary_conditions/concentration.md

```diff
@@ -0,0 +1,169 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.18.1
+kernelspec:
+  display_name: festim-workshop
+  language: python
+  name: python3
+---
+
+# Concentration #
+
+Boundary conditions (BCs) are essential to FESTIM simulations, as they describe the mathematical problem at the boundaries of the simulated domain. Read more about BCs _[here](https://festim.readthedocs.io/en/fenicsx/userguide/boundary_conditions.html)_.
+
+This tutorial goes over how to define concentration boundary conditions for hydrogen transport simulations.
+
+Objectives:
+* Define a fixed concentration BC
+* Choose which solubility law (Sieverts' or Henry's)
+* Solve a hydrogen transport problem with plasma implantation
+
++++
+
+## Defining fixed concentration ##
+
+BCs need to be assigned to surfaces using FESTIM's `SurfaceSubdomain` class. To define the concentration of a specific species, we can use `FixedConcentrationBC`:
+
+```{code-cell} ipython3
+from festim import FixedConcentrationBC, Species, SurfaceSubdomain
+
+boundary = SurfaceSubdomain(id=1)
+H = Species(name="Hydrogen")
+
+my_bc = FixedConcentrationBC(subdomain=boundary, value=10, species=H)
+```
+
+The imposed concentration can be dependent on space, time and temperature, such as:
+
+$$ 
+
+BC = 10 + x^2 + T(t+2)
+
+$$
+
+```{code-cell} ipython3
+my_custom_value = lambda x, t, T: 10 + x[0]**2 + T *(t + 2)
```

Comment:
We should also note that (x[0], x[1], x[2]) corresponds to (x, y, z) in a cartesian space